core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -119,36 +119,24 @@ public ThisT scanManifestsWith(ExecutorService executorService) {
 
   @Override
   public ThisT toBranch(String branch) {
-    throw new UnsupportedOperationException("Performing operations on a branch is currently not supported");
-  }
-
-  /***
-   * Will be used by snapshot producer operations to create a new ref if an invalid branch is passed
-   * @param branch ref name on which operation is to performed
-   */
-  protected void createNewRef(String branch) {
-    SnapshotRef branchRef = SnapshotRef.branchBuilder(this.current().currentSnapshot().snapshotId()).build();
-    TableMetadata.Builder updatedBuilder = TableMetadata.buildFrom(this.current());
-    updatedBuilder.setRef(branch, branchRef);
-    ops.commit(ops.current(), updatedBuilder.build());
```

Comment:
Removed this because in your change you do setBranchSnapshot and correctly pass in the target branch. That will create the branch if it does not exist.

Reply:
https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1489 code pointer